### PR TITLE
Refactor databases and make them work on Travis

### LINKED ISSUE #319: Make databases work on Travis
For example https://github.com/goavki/phenny/blob/master/modules/test/test_clock.py#L20

### PR DESCRIPTION
**Pull request by Androbin**
Fixes #319 
* `setup` methods are used consistently
* `db_path` guarantees the existence of the `.phenny` directory
* `read_db` and `write_db` take care of paths and encoding

### CODE REVIEW

**Comment by sushain97** [comment]
The build is failing, somewhat ironic.

**Comment by Androbin** [comment]
> The build is failing, somewhat ironic.

See #367 for the flawky test
Also, I marked this "work in progress" for a reason ;)

**Comment by Androbin** [comment]
@sushain97 Build is passing. Fixed and added `clock.py` tests as requested and they are doing fine.

**Comment by sushain97** [comment]
^ slightly conflicted, I believe.

File: modules/ethnologue.py [hunk 1]
```diff
@@ -141,8 +128,7 @@ def ethnologue(phenny, input):
 ethnologue.priority = 'low'
 
 def setup(phenny):
-    file = filename(phenny)
-    if os.path.exists(file):
+    try:
         read_ethnologue_codes(phenny)
-    else:
+    except:
```
Comment (sushain97) [inline modules/ethnologue.py]: Bare except?

File: modules/queue.py [hunk 2]
```diff
@@ -118,12 +106,12 @@ def queue(phenny, raw):
                         queue = raw.group(3).split(',')
                         queue = list(map(lambda x: x.strip(), queue))
                         phenny.queue_data[queue_name] = {'owner': owner, 'queue': queue}
-                        write_dict(filename(phenny), phenny.queue_data)
+                        write_db(phenny, 'queue', phenny.queue_data)
```
Comment (sushain97) [inline modules/queue.py]: Seems like this should be refactored into a function, the identical call is repeated 10+ times.

**Comment by sushain97** [comment]
@jonorthwash could you test this out with begiak to see if everything fails?

File: modules/ethnologue.py (hunk 1 above)
Comment (Androbin) [reply modules/ethnologue.py]: Okay, specified the types.

File: modules/queue.py (hunk 2 above)
Comment (Androbin) [reply modules/queue.py]: Yeah, but lots of slight variations make it difficult.
Naive refactoring would pass a dozen variables.
This should be a separate issue, I'll send a PR.

**Comment by Androbin** [comment]
@sushain97 @jonorthwash Do we yet know if this works?

**Comment by jonorthwash** [comment]
Does not seem to:

```
$ ./phenny -c ../.phenny/default.py
INFO: Setting up phenny
Traceback (most recent call last):
  File "./phenny", line 164, in <module>
    main()
  File "./phenny", line 161, in main
    run(config_module) # @@ thread this
  File "/home/begiak/phenny/__init__.py", line 74, in run
    t.run()
  File "/usr/lib/python3.5/threading.py", line 862, in run
    self._target(*self._args, **self._kwargs)
  File "/home/begiak/phenny/__init__.py", line 54, in run_phenny
    p = bot.Phenny(config)
  File "/home/begiak/phenny/bot.py", line 45, in __init__
    self.setup()
  File "/home/begiak/phenny/bot.py", line 84, in setup
    module.setup(self)
  File "/home/begiak/phenny/modules/remind.py", line 23, in setup
    phenny.remind_data = load_database(phenny)
  File "/home/begiak/phenny/modules/remind.py", line 17, in load_database
    return read_db(phenny, 'reminders')
  File "/home/begiak/phenny/tools.py", line 61, in read_db
    return pickle.load(f)
EOFError: Ran out of input
```

**Comment by Androbin** [comment]
@jonorthwash Seems like I missed an exception type.

**Comment by Androbin** [comment]
> exception `pickle.UnpicklingError`
> This exception is raised when there is a problem unpickling an object. Note that other exceptions may also be raised during unpickling, including (but not necessarily limited to) `AttributeError`, `EOFError`, `ImportError`, and `IndexError`.

Translation: The method might throw anything!
Seems like we need to put the catch-all back in...
https://docs.python.org/2/library/pickle.html#pickle.UnpicklingError

**Comment by Androbin** [comment]
@jonorthwash Could you try with the new exception handling?

**Comment by jonorthwash** [comment]
Nope.

```
$ ./phenny -vc ../.phenny/default.py
INFO: Setting up phenny
Traceback (most recent call last):
  File "/home/begiak/phenny/tools.py", line 67, in read_db
    return pickle.load(f)
EOFError: Ran out of input

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "./phenny", line 164, in <module>
    main()
  File "./phenny", line 161, in main
    run(config_module) # @@ thread this
  File "/home/begiak/phenny/__init__.py", line 74, in run
    t.run()
  File "/usr/lib/python3.5/threading.py", line 862, in run
    self._target(*self._args, **self._kwargs)
  File "/home/begiak/phenny/__init__.py", line 54, in run_phenny
    p = bot.Phenny(config)
  File "/home/begiak/phenny/bot.py", line 45, in __init__
    self.setup()
  File "/home/begiak/phenny/bot.py", line 84, in setup
    module.setup(self)
  File "/home/begiak/phenny/modules/remind.py", line 23, in setup
    phenny.remind_data = load_database(phenny)
  File "/home/begiak/phenny/modules/remind.py", line 17, in load_database
    return read_db(phenny, 'reminders')
  File "/home/begiak/phenny/tools.py", line 70, in read_db
    raise GrumbleError()
tools.GrumbleError
```

**Comment by Androbin** [comment]
Yeah, that module didn't catch the exception. Gimme a moment.

**Comment by Androbin** [comment]
Ups, forgot the fallback for that module.

**Comment by Androbin** [comment]
Okay, let me check:
* All `pickle.load` exceptions are re-thrown as `GrumbleError`
* All `read_db` exceptions (`GrumbleError`, `ResourceWarning`) are handled
* All imports are updated

@jonorthwash Let's try one more time.

**Comment by jonorthwash** [comment]
Yeah, it seems to work.  Should we merge @sushain97 ?

**Comment by sushain97** [comment]
Looks good to me.

**Comment by Androbin** [comment]
@shardulc FYI

**Pull request merged**